src/lightning/pytorch/strategies/deepspeed.py

```diff
@@ -343,12 +329,10 @@ def setup_distributed(self) -> None:
         reset_seed()
         self.set_world_ranks()
         self._init_deepspeed_distributed()
-        if not self._config_initialized:
-            self._format_config()
-            self._config_initialized = True
```

Comment:
Moved this part to a method `init_config_if_needed`. This was too early here, so moved the formatting of the config to `model_sharded_context` below